src/app/main/projects/project-detail/project-detail.component.ts

```diff
@@ -0,0 +1,34 @@
+import { Component, OnInit } from '@angular/core';
+import {projects} from "../all-projects/projects.model";
+import {Router, ActivatedRoute, Params} from "@angular/router";
+
+@Component({
+  selector: 'app-project-detail',
+  templateUrl: './project-detail.component.html',
+  styleUrls: ['./project-detail.component.scss']
+})
+export class ProjectDetailComponent implements OnInit {
+
+  project;
+
+  constructor(private router: Router, private activatedRouter : ActivatedRoute) { }
+
+  ngOnInit() {
+
+    this.activatedRouter.params.forEach((params : Params) =>{
+
+      let id = +params["id"];
+
+      this.project = projects.find(x  => x.id == id);
+
```

Comment:
delete unnecesarry rows, add spaces, use `const` instead of `let` when the variable will not be changed,
use strict equal operator `===`, use single quotes `''` in .ts files and double quotes `""` in .html files
```
ngOnInit() {
    this.activatedRouter.params.forEach((params : Params) => {
      const id = +params["id"];
      this.project = projects.find(x => x.id === id);
}
```